### PR TITLE
Restrict training annotation bounding boxes to volume layer bbox

### LINKED ISSUE #9676: Restrict training annotation boxes to volume layer bounding box
The training doesn't work if a bounding box in a training annotation is outside of the volume layer's bounding box. This should ideally be caught in the UI already, i.e., when a user wants to start a training, there should be an error message if one of the boxes it outside the volume layer bbox.

### PR DESCRIPTION
**Pull request by hotzenklotz**
This PR enforces a VX rule for training to restrict training annotation bounding boxes to volume layer bbox.

### Steps to test:
- Open an annotation with a volume layer 
- Create at least one user bounding box that extends outside of that volume layer bounding box
- Open the AI training drawer ("Train a model").
- The affected annotation card should now show a red error listing the offending bounding boxes, and the "Start training" button should be disabled.
- Moving/shrinking the bounding boxes back inside the volume layer bbox clears the error and re-enables the button.

Screenshot 2026-07-06 at 16-27-25

### Issues:
- fixes #9676

------
- [x] Added changelog entry (create a `$PR_NUMBER.md` file in `unreleased_changes` or use `./tools/create-changelog-entry.py`)
- [ ] Added migration guide entry if applicable (edit the same file as for the changelog)
- [ ] Updated documentation if applicable
- [ ] Adapted wk-libs python client if relevant API parts change
- [x] Removed dev-only changes like prints and application.conf edits
- [x] Considered common edge cases
- [ ] Needs datastore update after deployment

### CODE REVIEW

**Comment by hotzenklotz** [comment]
@MichaelBuessemeyer I am unsure if the unit test actual adds value... Let me know your thoughts

File: frontend/javascripts/viewer/view/ai_jobs/utils.ts [hunk 1]
```diff
@@ -37,6 +38,49 @@ export function getBoundingBoxesForLayers(layers: APIDataLayer[]): UserBoundingB
...
+  );
+  if (layer?.tracingId == null) {
+    return null;
+  }
+  const tracing = volumeTracings.find((t) => t.tracingId === layer.tracingId);
+  if (tracing?.boundingBox == null) {
+    return null;
+  }
+  return new BoundingBox(tracing.boundingBox);
+}
```
Comment (MichaelBuessemeyer) [inline frontend/javascripts/viewer/view/ai_jobs/utils.ts:66]: for a better placement maybe put this into the `annotation_accessor.ts` and the function below into the bounding box class (`frontend/javascripts/viewer/model/bucket_data_handling/bounding_box.ts`) and for the test create bounding box objects?

What do you think?

File: frontend/javascripts/test/viewer/ai_jobs_utils.spec.ts
Comment (MichaelBuessemeyer) [inline frontend/javascripts/test/viewer/ai_jobs_utils.spec.ts:1]: I know that tests are good but they also add code to the codebase we need to maintain.
I discussed this with philipp in the past and we came to the agreement that small not so important features of wk do not need such tests. As I'd say that both functions are relatively straight forward implementation wise they do not need any testing. So I'd prefer to remove these tests

File: frontend/javascripts/viewer/view/ai_jobs/utils.ts (hunk 1 above)
Comment (MichaelBuessemeyer) [reply frontend/javascripts/viewer/view/ai_jobs/utils.ts:66]: on second thought `getGroundTruthLayerBoundingBox` might not perfectly match the `annotation_accessor.ts` as this usually handles the `StoreAnnotation` type and not the `APIAnnotation` type 🤔. I'd say do what you prefer :D

**Review by MichaelBuessemeyer — APPROVED** [review]
> I am unsure if the unit test actual adds value... Let me know your thoughts

Oh interesting: I just thought the same here. So lets get rid of them 🎉 

Testing worked well. Thanks for adding this validation step 🙏 

I added one more comment regarding the location of the two new helper functions
